pkg/controller/core.oam.dev/v1alpha2/application/application_controller.go

```diff
@@ -33,27 +33,30 @@ import (
 	"k8s.io/klog/v2"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
-	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 
 	"github.com/oam-dev/kubevela/apis/core.oam.dev/common"
 	"github.com/oam-dev/kubevela/apis/core.oam.dev/v1beta1"
 	velatypes "github.com/oam-dev/kubevela/apis/types"
 	"github.com/oam-dev/kubevela/pkg/appfile"
 	core "github.com/oam-dev/kubevela/pkg/controller/core.oam.dev"
+	"github.com/oam-dev/kubevela/pkg/controller/core.oam.dev/v1alpha2/application/dispatch"
 	"github.com/oam-dev/kubevela/pkg/cue/packages"
+	"github.com/oam-dev/kubevela/pkg/oam"
 	"github.com/oam-dev/kubevela/pkg/oam/discoverymapper"
 	oamutil "github.com/oam-dev/kubevela/pkg/oam/util"
 	"github.com/oam-dev/kubevela/pkg/utils/apply"
 )
 
-// RolloutReconcileWaitTime is the time to wait before reconcile again an application still in rollout phase
 const (
-	RolloutReconcileWaitTime      = time.Second * 3
-	resourceTrackerFinalizer      = "resourceTracker.finalizer.core.oam.dev"
 	errUpdateApplicationStatus    = "cannot update application status"
 	errUpdateApplicationFinalizer = "cannot update application finalizer"
 )
 
+const (
+	resourceTrackerFinalizer = "latestResourceTracker.finalizer.core.oam.dev"
+	onlyRevisionFinalizer    = "onlyRevision.finalizer.core.oam.dev"
```

Comment:
let's explan the usage of the finalizer here

Reply:
Key names should be all lowercase, with words separated by dashes instead of camelCase
For instance, prefer foo.kubernetes.io/foo-bar over foo.kubernetes.io/fooBar, prefer desired-replicas over DesiredReplicas